### PR TITLE
GUACAMOLE-663: Set pthread stack size for guacd

### PR DESCRIPTION
**Pull request by necouchman**
Applying the patch provided on the Jira issue.

### CODE REVIEW

File: src/guacd/daemon.c [hunk 1]
```diff
@@ -17,6 +17,8 @@
  * under the License.
  */
 
+#define _GNU_SOURCE 1
```
Comment (mike-jumper) [inline src/guacd/daemon.c:20]: Which function is this required for?

Comment (necouchman) [reply src/guacd/daemon.c:20]: Required for `pthread_setattr_default_np()`. I'm happy to either add a comment to that effect, or use an alternative function?

File: src/guacd/daemon.c [hunk 2]
```diff
@@ -275,6 +282,12 @@ int main(int argc, char* argv[]) {
     /* General */
     int retval;
 
+    /* Set default stack size of 8MB */
+    pthread_attr_t default_pthread_attr;
+    pthread_attr_init(&default_pthread_attr);
+    pthread_attr_setstacksize(&default_pthread_attr, GUACD_THREAD_STACK_SIZE);
+    pthread_setattr_default_np(&default_pthread_attr);
```
Comment (mike-jumper) [inline src/guacd/daemon.c:289]: I don't think we should rely on `_GNU_SOURCE` causing `pthread_setattr_default_np()` to be defined. There's no requirement that a libc implementation behave differently with `_GNU_SOURCE` defined, nor that it provide `pthread_setattr_default_np()` to those defining it.

Instead, we should:

1. Use configure tests to check whether `pthread_setattr_default_np()` is available, setting `_GNU_SOURCE` if necessary.
2. Make these calls only if `pthread_setattr_default_np()` is available.

If it's possible to check whether setting `LDFLAGS` options is sufficient to establish the expected stack size, we might want to do that, as well. Setting `LDFLAGS` is the mechanism we use to achieve this within the current Alpine build:

https://github.com/apache/guacamole-server/blob/47b9360d4678bb8e60f0850090ccade666255069/src/guacd-docker/bin/build-all.sh#L34-L36

Comment (necouchman) [reply src/guacd/daemon.c:329]: Okay, this should be addressed via rebase.

File: src/guacd/daemon.c [hunk 3]
```diff
@@ -320,6 +321,14 @@ int main(int argc, char* argv[]) {
     /* General */
     int retval;
 
+#ifdef HAVE_DECL_PTHREAD_SETATTR_DEFAULT_NP
+    /* Set default stack size of 8MB */
```
Comment (jmuehlner) [inline src/guacd/daemon.c:325]: I don't love having the value of the constant specified here in the constant - it's an opportunity for things to get out of sync if somebody later updates the value of `GUACD_THREAD_STACK_SIZE`.

**Comment by jmuehlner** [comment]
This PR looks to contain a few unrelated commits - bad rebase?

File: src/guacd/daemon.c (hunk 3 above)
Comment (necouchman) [reply src/guacd/daemon.c]: Makes sense. I've removed that part of the comment.

**Comment by necouchman** [comment]
> This PR looks to contain a few unrelated commits - bad rebase?

Whoops, yeah, had rebased it against `staging/1.6.0` there before we decided to remove it, and it got a few extra commits. Should be good, now.

**Comment by jmuehlner** [comment]
LGTM

**Pull request merged**